functions/src/timetables/__tests__/Timetable.controller.test.ts

```diff
@@ -0,0 +1,86 @@
+import { mocked } from 'ts-jest/utils';
+import { addTimetable, getTimetable } from '../Timetable.service';
+
+import * as request from 'supertest';
+import * as express from 'express';
+import { RegisterRoutes } from '../../../build/routes';
+
+jest.mock('../Timetable.service.ts');
+
+const mockGetTimetable = mocked(getTimetable);
+const mockAddTimetable = mocked(addTimetable);
+
+const app = express();
+RegisterRoutes(app);
+
+const { get, post } = request(app);
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const timetable: any = {
+  slug: 'abc123',
+  term: '202109',
+  courses: [
+    {
+      subject: 'ACAN',
+      code: '225',
+      pid: 'ByS23Pp7E',
+      lecture: 'A02',
+      lab: 'B02',
+      color: '#48BB78',
```

Comment:
Does validation run on these attributes? It would be good to test if weird values cannot be set in these.

Reply:
it's mocked so don't think so